### PR TITLE
Documentation and blog updates

### PR DESCRIPTION
**Pull request by segeljakt**

### CODE REVIEW

File: arc-docs/src/examples/signal-smoothing.md [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
 ## Implementation
 
-```
-fun query(stream) {
+```arc-lang
+def query(stream) {
     stream
       .swindow(5, 1, fun(v): (v[0] + 2*v[1] + 4*v[2] + 2*v[3] + v[4]) / 10.0)
```
Comment (frej) [inline arc-docs/src/examples/signal-smoothing.md]: Cool, a FIR filter.

**Review by frej — APPROVED** [review]
LGTM

File: arc-docs/src/tour/tasks.md [hunk 2]
```diff
@@ -1,26 +1,150 @@
...
-    loop {
-        val x = receive s0;
-        val y = receive s1;
-        emit f(x, y) in s2;
+
+In the above code, `map` is a task which takes a stream on an input channel `i`, a function `f`, and produces a stream on an output channel `o`. The execution of the task enters an infinite loop which receives events from `i` using the `receive` operator, applies function `f`, and emits the result to `o` using the `!` operator. The `main` function instantiates the task using function application syntax.
+
+## Capabilities
+
+All channels have a *capability*. Data can either be pushed into a channel (push-capability) or pulled out of it (pull-capability), but not both. Tasks which push data are referred to as "*producers*" and tasks which pull data are referred to as "*consumers*". It is possible for a task to simultaneously be both a producer and a consumer, but not for the same channel. We refer to channels with pull-capabilities as streams (`Stream[T]`) and channels with push-capabilities as drains (`Drain[T]`). To make the idea more clear, consider this fully type annotated example:
```
[the remark this answers is not in the document]
Comment (segeljakt) [reply arc-docs/src/tour/tasks.md:13]: Hmm, I did not know. We could rename it to "channel directions" like in Go.

File: arc-lang/examples/parallelism.arc [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# XFAIL: *
+# RUN: arc -o %t run %s -- -rustinclude %s.rust-tests
+# RUN: arc -o %t-canon run %s -- -rustinclude %s.rust-tests -canonicalize
```
[the remark this answers is not in the document]
Comment (segeljakt) [reply arc-lang/examples/parallelism.arc:3]: Oops, I forgot to create these files

File: arc-docs/src/tour/tasks.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (frej) [reply arc-docs/src/tour/tasks.md:13]: If Go has an established term, using it is instead of overloading something else is much better.

**Pull request merged**